diff/diff.go

```diff
@@ -58,7 +58,7 @@ type Applier interface {
 	// implementation and content descriptor. For example, in the common
 	// case the descriptor is a file system difference in tar format,
 	// that tar would be applied on top of the mounts.
-	Apply(ctx context.Context, desc ocispec.Descriptor, mount []mount.Mount) (ocispec.Descriptor, error)
+	Apply(ctx context.Context, desc ocispec.Descriptor, mount []mount.Mount, options ...map[string]string) (ocispec.Descriptor, error)
```

Comment:
I think it is fine to just use `Opt` here and add a `WithOption(string, string)` to set it. Could these same options be necessary on compare in the future?

Reply:
```
// Opt is used to configure a diff operation
type Opt func(*Config) error
```
How would we pass this function via gRPC ?

Reply:
Normally the client just resolves the opts, sets the appropriate parameters in the GRPC request, then the server handler reads them and recreates the opts. GRPC always has to have the options broken out in the struct since we want them to be strongly typed.

Reply:
My bad. Had been looking for examples for existing code that does this but had looked at the wrong place. Fixed now.